--- a/bundles/AdminBundle/Controller/Admin/Document/PageController.php
+++ b/bundles/AdminBundle/Controller/Admin/Document/PageController.php
@@ -172,6 +172,8 @@
             $page->setMetaData($metaData);
         }
 
+        $this->applySchedulerDataToElement($request, $page);
+
         // only save when publish or unpublish
         if (($request->get('task') == 'publish' && $page->isAllowed('publish')) || ($request->get('task') == 'unpublish' && $page->isAllowed('unpublish'))) {
             $this->setValuesToDocument($request, $page);
@@ -195,7 +197,13 @@
                     'versionCount' => $page->getVersionCount(),
                 ],
             ]);
-        } elseif ($page->isAllowed('save')) {
+        } elseif ($request->get('task') == 'scheduler') {
+            if ($page->isAllowed('settings')) {
+                $page->saveScheduledTasks();
+
+                return $this->adminJson(['success' => true]);
+            }
+        }  elseif ($page->isAllowed('save')) {
             $this->setValuesToDocument($request, $page);
 
             $page->saveVersion();
--- a/bundles/AdminBundle/Controller/Admin/Document/SnippetController.php
+++ b/bundles/AdminBundle/Controller/Admin/Document/SnippetController.php
@@ -167,6 +167,13 @@
                 ],
                 'treeData' => $treeData,
             ]);
+        } elseif ($request->get('task') == 'scheduler') {
+            if ($snippet->isAllowed('settings')) {
+                $this->setValuesToDocument($request, $snippet);
+                $snippet->saveScheduledTasks();
+
+                return $this->adminJson(['success' => true]);
+            }
         } elseif ($snippet->isAllowed('save')) {
             $this->setValuesToDocument($request, $snippet);

