--- a/bundles/WordExportBundle/src/Controller/TranslationController.php
+++ b/bundles/WordExportBundle/src/Controller/TranslationController.php
@@ -68,16 +68,17 @@
                 $element = \Pimcore\Model\Element\Service::getElementById($el['type'], $el['id']);
                 $output = '';
 
-                // check supported types (subtypes)
-                if (!in_array($element->getType(), ['page', 'snippet', 'email', 'object'])) {
+                // check supported types (subtypes) and if user has view permission
+                if (
+                    !in_array($element?->getType(), ['page', 'snippet', 'email', 'object'], true) ||
+                    !$element->isAllowed('view')
+                ) {
                     continue;
                 }
 
-                if ($element instanceof ElementInterface) {
-                    $output .= '<h1 class="element-headline">' . ucfirst(
-                        $element->getType()
-                    ) . ' - ' . $element->getRealFullPath() . ' (ID: ' . $element->getId() . ')</h1>';
-                }
+                $output .= '<h1 class="element-headline">' . ucfirst(
+                    $element->getType()
+                ) . ' - ' . $element->getRealFullPath() . ' (ID: ' . $element->getId() . ')</h1>';
 
                 if ($element instanceof PageSnippet) {
                     if ($element instanceof Page) {
