--- src/Resources/public/js/searchConfig/resultPanel.js	2023-10-19 12:56:39
+++ src/Resources/public/js/searchConfig/resultPanel.js	2023-10-18 12:20:42
@@ -428,7 +428,7 @@
             for (var i = 0; i < selectedRows.length; i++) {
                 jobs.push(selectedRows[i].get("id"));
             }
-            this.batchOpen(columnIndex, jobs, append, remove);
+            this.batchOpen(columnIndex, jobs, append, remove, onlySelected);
 
         } else {
 
@@ -460,7 +460,7 @@
                 success: function (columnIndex, response) {
                     var rdata = Ext.decode(response.responseText);
                     if (rdata.success && rdata.jobs) {
-                        this.batchOpen(columnIndex, rdata.jobs, append, remove);
+                        this.batchOpen(columnIndex, rdata.jobs, append, remove, onlySelected);
                     }
 
                 }.bind(this, columnIndex)
