Subject: [PATCH] Fix for scheduling CMS pages
---
Index: bundles/AdminBundle/Controller/Admin/Document/PageController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bundles/AdminBundle/Controller/Admin/Document/PageController.php b/bundles/AdminBundle/Controller/Admin/Document/PageController.php
--- a/bundles/AdminBundle/Controller/Admin/Document/PageController.php	(revision 9a33afacc268d9779bf6419639e8c208355e50c0)
+++ b/bundles/AdminBundle/Controller/Admin/Document/PageController.php	(date 1675418918231)
@@ -172,6 +172,8 @@
             $page->setMetaData($metaData);
         }
 
+        $this->applySchedulerDataToElement($request, $page);
+
         // only save when publish or unpublish
         if (($request->get('task') == 'publish' && $page->isAllowed('publish')) || ($request->get('task') == 'unpublish' && $page->isAllowed('unpublish'))) {
             $this->setValuesToDocument($request, $page);
@@ -195,7 +197,13 @@
                     'versionCount' => $page->getVersionCount(),
                 ],
             ]);
-        } elseif ($page->isAllowed('save')) {
+        } elseif ($request->get('task') == 'scheduler') {
+            if ($page->isAllowed('settings')) {
+                $page->saveScheduledTasks();
+
+                return $this->adminJson(['success' => true]);
+            }
+        }  elseif ($page->isAllowed('save')) {
             $this->setValuesToDocument($request, $page);
 
             $page->saveVersion();
