--- src/Resources/public/js/searchConfig/resultPanel.js	2023-02-01 16:22:52
+++ src/Resources/public/js/searchConfig/resultPanel.js	2023-06-20 10:04:09
@@ -81,7 +81,7 @@
     updateGrid: function (classId) {
         this.classId = classId;
         var classStore = pimcore.globalmanager.get("object_types_store");
-        var classRecord = classStore.findRecord("id", this.classId);
+        var classRecord = classStore.findRecord("id", this.classId, 0, false, false, true);
         if (classRecord) {
             this.selectedClass = classRecord.data.text;
 
