--- resultPanel.js	2023-09-19 12:02:21.440728401 +0200
+++ resultPanel2.js	2023-09-19 12:02:12.340730600 +0200
@@ -90,7 +90,7 @@

                 Ext.Ajax.request({
                     url: "/admin/object-helper/grid-get-column-config",
-                    params: {name: this.selectedClass, gridtype: "grid"},
+                    params: {name: this.selectedClass, gridtype: "grid", searchType: "folder"},
                     success: function (response) {
                         response = Ext.decode(response.responseText);
                         var settings = response.settings || {};
@@ -105,7 +105,7 @@
             } else {
                 Ext.Ajax.request({
                     url: "/admin/object-helper/grid-get-column-config",
-                    params: {name: this.selectedClass, gridtype: "grid"},
+                    params: {name: this.selectedClass, gridtype: "grid", searchType: "folder"},
                     success: this.createGrid.bind(this, false)
                 });
             }
@@ -120,7 +120,7 @@
                 id: this.classId,
                 gridtype: "grid",
                 gridConfigId: this.settings ? this.settings.gridConfigId : null,
-                searchType: this.searchType
+                searchType: this.searchType ? this.searchType : "folder"
             },
             success: this.createGrid.bind(this, false)
         });
@@ -562,7 +562,7 @@
             }.bind(this), function (data) {
                 Ext.Ajax.request({
                     url: "/admin/object-helper/grid-get-column-config",
-                    params: {name: this.selectedClass, gridtype: "grid"},
+                    params: {name: this.selectedClass, gridtype: "grid", searchType: "folder"},
                     success: this.createGrid.bind(this, false)
                 });
             }.bind(this)
